py-polars/polars/expr/datetime.py

```diff
@@ -1423,6 +1423,52 @@ def replace_time_zone(
             self._pyexpr.dt_replace_time_zone(time_zone, ambiguous._pyexpr)
         )
 
+    def convert_to_local_time_zone(
+        self,
+        local_time_zone: IntoExpr,
+    ) -> Expr:
+        """
+        Converts to local timezone and then replaces timezone with None.
+
+        Parameters
+        ----------
+        local_time_zone
+            Time zone for the `Datetime` expression.
+
+        Returns
+        -------
+        Expr
+            Expression of data type :class:`DateTime`.
+
+        Examples
+        --------
+        >>> from datetime import datetime
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "date_col": [datetime(2020, 10, 10)] * 3,
+        ...         "timezone": ["Europe/London", "Africa/Kigali", "America/New_York"],
+        ...     }
+        ... ).with_columns(pl.col("date_col").dt.replace_time_zone("UTC"))
+        >>> df.with_columns(
+        ...     pl.col("date_col")
+        ...     .dt.convert_to_local_time_zone("timezone")
```

Comment:
is this the API we want? how about `.dt.convert_to_local_time_zone(pl.col("timezone"))`, and if you pass a string, then that's the time zone?

Reply:
hmm true, thanks